pkg/auth/dependency/userverify/provider.go

```diff
@@ -144,16 +149,32 @@ func (provider *providerImpl) markUserVerified(
 
 	// Update user
 	authInfo.VerifyInfo[verifyCode.LoginID] = true
-	authInfo.Verified = IsUserVerified(
+	if err = provider.UpdateVerificationState(authInfo, authStore, principals); err != nil {
+		return
+	}
+
+	return
+}
+
+func (provider *providerImpl) UpdateVerificationState(
+	authInfo *authinfo.AuthInfo,
+	authStore authinfo.Store,
+	principals []*password.Principal,
+) error {
+	isVerified := IsUserVerified(
```

Comment:
I think we need to remove the login ID from authInfo.VerifyInfo if the login ID is removed or updated.

Reply:
I suppose VerifyInfo would contains the email addresses that the user possessed (i.e. verified emails), therefore the user would be verified if the verified email is removed and added as login ID again.

Reply:
This is unspecified in the feature spec. We should discuss this with the team. Maybe we open a PR to update the related spec and continue the discussion there?

Reply:
The user verification feature is not specified (I opened feature#336 for tracking but it is closed). I'll add some concerns to discuss there.